### PR TITLE
REF: re-use maybe_cast_result in Series.combine

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/dtypes/cast.py [hunk 1]
```diff
@@ -375,7 +374,11 @@ def trans(x):
 
 
 def maybe_cast_result(
-    result: ArrayLike, obj: Series, numeric_only: bool = False, how: str = ""
+    result: ArrayLike,
+    dtype: DtypeObj,
+    numeric_only: bool = False,
+    how: str = "",
+    same_dtype: bool = True,
```
Comment (jreback) [inline pandas/core/dtypes/cast.py:381]: i would instead call this `casting`, in a similar vein to: `np.astype` (so maybe just all 'no' | 'equiv' or similar here

Comment (jreback) [reply pandas/core/dtypes/cast.py:381]: could also make an enum for this

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/dtypes/cast.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/cast.py:381]: bool is a lot clearer than an enum, doesnt require the reader to go find the enum definition

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks, this is a nice cleanup

**Pull request merged**